From 9a384deba7fd4da310aefcc56e6173c98b9db2af Mon Sep 17 00:00:00 2001
From: Matiss <matissp1302@gmail.com>
Date: Tue, 21 Apr 2026 10:47:02 +0200
Subject: [PATCH 1/2] fix: typo fix

---
 Correlator/DeviceInstance.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Correlator/DeviceInstance.cc b/Correlator/DeviceInstance.cc
index 86a4160..227113e 100644
--- a/Correlator/DeviceInstance.cc
+++ b/Correlator/DeviceInstance.cc
@@ -62,7 +62,7 @@ DeviceInstance::DeviceInstance(CorrelatorPipeline &pipeline, unsigned deviceNr)
     return TCC(device, ps);
   })),
 
-  devCorrectedData((size_t) ps.nrChannelsPerSubband() * ps.nrSamplesPerChannel() * ps.nrStations() * ps.nrPolarizations() * sizeof(__half2))),
+  devCorrectedData((size_t) ps.nrChannelsPerSubband() * ps.nrSamplesPerChannel() * ps.nrStations() * ps.nrPolarizations() * sizeof(__half2)),
   devInputBuffer((size_t) ps.nrStations() * ps.nrPolarizations() * (ps.nrSamplesPerChannel() + NR_TAPS - 1) * ps.nrChannelsPerSubbandBeforeFilter() * ps.nrBytesPerRealSample()),
   
   filters([&] {
-- 
2.54.0

