From 47b434fe5a32371e321332f1174b6335ec2e1d6a Mon Sep 17 00:00:00 2001
From: skalndrausis <janis.akmenskalns@gmail.com>
Date: Wed, 13 May 2026 11:42:42 +0000
Subject: [PATCH] fix: removed the 1024 multiplier from TimeStamp class

---
 Common/TimeStamp.h | 30 +++++++-----------------------
 1 file changed, 7 insertions(+), 23 deletions(-)

diff --git a/Common/TimeStamp.h b/Common/TimeStamp.h
index 889448d..6e0deea 100644
--- a/Common/TimeStamp.h
+++ b/Common/TimeStamp.h
@@ -82,41 +82,26 @@ inline TimeStamp::TimeStamp(int64_t time, unsigned clockSpeed)
 
 inline TimeStamp::TimeStamp(unsigned seqId, unsigned blockId, unsigned clockSpeed)
 :
-#ifdef EVEN_SECOND_HAS_MORE_SAMPLES
-  time(((int64_t) seqId * clockSpeed + 512) / 1024 + blockId),
-#else
-  time(((int64_t) seqId * clockSpeed) / 1024 + blockId),
-#endif
+  time((int64_t) seqId * clockSpeed + blockId),
   clockSpeed(clockSpeed)
 {
 }
 
 inline TimeStamp &TimeStamp::setStamp(unsigned seqId, unsigned blockId)
 {
-#ifdef EVEN_SECOND_HAS_MORE_SAMPLES
-  time = ((int64_t) seqId * clockSpeed + 512) / 1024 + blockId;
-#else
-  time = ((int64_t) seqId * clockSpeed) / 1024 + blockId;
-#endif
+  time = (int64_t) seqId * clockSpeed + blockId;
   return *this;
 }
 
 inline unsigned TimeStamp::getSeqId() const
 {
-#ifdef EVEN_SECOND_HAS_MORE_SAMPLES
-  return (unsigned) (1024 * time / clockSpeed);
-#else
-  return (unsigned) ((1024 * time + 512) / clockSpeed);
+  return (unsigned) (time / clockSpeed);
 #endif
 }
 
 inline unsigned TimeStamp::getBlockId() const
 {
-#ifdef EVEN_SECOND_HAS_MORE_SAMPLES
-  return (unsigned) (1024 * time % clockSpeed / 1024);
-#else
-  return (unsigned) ((1024 * time + 512) % clockSpeed / 1024);
-#endif
+  return (unsigned) (time % clockSpeed);
 }
 
 template <typename T> inline TimeStamp &TimeStamp::operator += (T increment)
@@ -184,16 +169,15 @@ inline TimeStamp::operator int64_t () const
 
 inline TimeStamp::operator double () const
 {
-  return (double) 1024.0 * time / clockSpeed;
+  return (double) time / clockSpeed;
 }
 
 inline TimeStamp::operator struct timespec () const
 {
-  int64_t	  ns = (int64_t) (time * 1024 * 1e9 / clockSpeed);
   struct timespec ts;
 
-  ts.tv_sec  = ns / 1000000000ULL;
-  ts.tv_nsec = ns % 1000000000ULL;
+  ts.tv_sec  = time / clockSpeed;
+  ts.tv_nsec = (long) ((time % clockSpeed) * 1000000000LL / clockSpeed);
 
   return ts;
 }
-- 
2.54.0

